src/code/z_message.c

```diff
@@ -1,24 +1,167 @@
+    D_801F6B20 = 0xA;
+}
+
+s32 Message_ShouldAdvance(GlobalContext* globalCtx) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    Input* controller = CONTROLLER1(globalCtx);
+
+    if ((msgCtx->unk12020 == 0x10) || (msgCtx->unk12020 == 0x11)) {
+        if (CHECK_BTN_ALL(controller->press.button, BTN_A)) {
+            play_sound(NA_SE_SY_MESSAGE_PASS);
+        }
+        return CHECK_BTN_ALL(controller->press.button, BTN_A);
+    } else {
+        if (CHECK_BTN_ALL(controller->press.button, BTN_A) || CHECK_BTN_ALL(controller->press.button, BTN_B) ||
+            CHECK_BTN_ALL(controller->press.button, BTN_CUP)) {
+            play_sound(NA_SE_SY_MESSAGE_PASS);
+        }
+        return CHECK_BTN_ALL(controller->press.button, BTN_A) || CHECK_BTN_ALL(controller->press.button, BTN_B) ||
+               CHECK_BTN_ALL(controller->press.button, BTN_CUP);
+    }
+}
+
+s32 Message_ShouldAdvanceSilent(GlobalContext* globalCtx) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    Input* controller = CONTROLLER1(globalCtx);
+
+    if (msgCtx->unk12020 == 0x10 || msgCtx->unk12020 == 0x11) {
+        return CHECK_BTN_ALL(controller->press.button, BTN_A);
+    } else {
+        return CHECK_BTN_ALL(controller->press.button, BTN_A) || CHECK_BTN_ALL(controller->press.button, BTN_B) ||
+               CHECK_BTN_ALL(controller->press.button, BTN_CUP);
+    }
+}
+
+void func_801477B4(GlobalContext* globalCtx) {
+    MessageContext* msgCtx;
+
+    msgCtx = &globalCtx->msgCtx;
+    if (globalCtx->msgCtx.unk11F10 != 0) {
+        msgCtx->unk12023 = 2;
+        msgCtx->msgMode = 0x43;
+        msgCtx->unk12020 = 0;
+        play_sound(NA_SE_PL_WALK_GROUND - SFX_FLAG);
+    }
+}
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80147818.s")
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80147F18.s")
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80148558.s")
 
+#ifdef NON_MATCHING
+void func_80148B98(GlobalContext* globalCtx, u8 arg1) {
+    static s16 held = 0;
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    Input* curInput = CONTROLLER1(globalCtx);
+
+    if ((curInput->rel.stick_y > 29) && held == 0) {
+        held = 1;
+        msgCtx->choiceIndex--;
+        if (msgCtx->choiceIndex > 128) {
+            msgCtx->choiceIndex = 0;
+        } else {
+            play_sound(NA_SE_SY_CURSOR);
+        }
+        return;
+    }
+
+    else if ((curInput->rel.stick_y < -29) && held == 0) {
+        held = 1;
+        msgCtx->choiceIndex++;
+        if (msgCtx->choiceIndex > arg1) {
+            msgCtx->choiceIndex = arg1;
+        } else {
+            play_sound(NA_SE_SY_CURSOR);
+        }
+        return;
+    } else {
+        if (ABS_ALT(curInput->rel.stick_y) < 30) {
+            held = 0;
+        }
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80148B98.s")
+#endif
+
+void func_80148CBC(GlobalContext* globalCtx, UNK_PTR puParm2, u8 arg2) {
+    MessageContext* msgCtx;
```

Comment:
```suggestion
    MessageContext* msgCtx = &globalCtx->msgCtx;
```

Reply:
done in https://github.com/zeldaret/mm/commit/eaa4f44462da18c9c81ba828fca65470abf3b526